pkg/database/connection.go

```diff
@@ -1,58 +1,40 @@
-// Copyright KubeArchive Authors
-// SPDX-License-Identifier: Apache-2.0
-
 package database
 
 import (
-	"fmt"
-	"os"
-)
-
-type EnvVar string
-
-const (
-	connStrTpl         = "user=%s password=%s dbname=%s host=%s port=%s sslmode=disable"
-	dbConnectionErrStr = "Could not create database connection string: %s must be set"
+	"database/sql"
+	"log"
+	"time"
 
-	DbNameEnvVar     EnvVar = "POSTGRES_DB"
-	DbUserEnvVar     EnvVar = "POSTGRES_USER"
-	DbPasswordEnvVar EnvVar = "POSTGRES_PASSWORD" // #nosec G101 not a password
-	DbUrlEnvVar      EnvVar = "POSTGRES_URL"
-	DbPortEnvVar     EnvVar = "POSTGRES_PORT"
+	"github.com/XSAM/otelsql"
+	"github.com/avast/retry-go/v4"
 )
 
-// Returns the value of the environment variable dbEnvVar. If dbEnvVar is not set, readDbEnvVar returns an error
-func readDbEnvVar(dbEnvVar EnvVar) (string, error) {
-	envVarVal, exists := os.LookupEnv(string(dbEnvVar))
-	if !exists {
-		return "", fmt.Errorf(dbConnectionErrStr, string(dbEnvVar))
-	}
-	return envVarVal, nil
+type connectionData struct {
+	driver           string
+	connectionString string
```

Comment:
This struct feels a little bit useless. When we use it we are initializing it, we are passing these two parameters, just to call `establishConnection()` on it. What about having a  `establishConnection(driverName, connectionString) (*sql.DB, error)` function and call that instead?

Reply:
yes that makes sense

Reply:
Done